src/coreclr/jit/codegenarm64.cpp

```diff
@@ -3585,7 +3584,7 @@ void CodeGen::genCodeForCompare(GenTreeOp* tree)
         assert(!op1->isContained());
         assert(op1Type == op2Type);
 
-        if (op2->IsIntegralConst(0))
```

Comment:
@EgorBo So, effectively, this was always dead code, right?

Reply:
@echesakovMSFT yep